moshi/moshi/modules/transformer.py

```diff
@@ -466,12 +475,23 @@ def forward(self, query: torch.Tensor, key: torch.Tensor, value: torch.Tensor):
             offset = state.offset
             offset_cpu = state.offset_cpu
 
-        projected = apply_weights_per_step(
-            self.in_projs, self.weights_per_step_schedule, query, offset_cpu)
+        if self.cross_attention:
+            assert state is None
+            assert len(self.in_projs) == 1
+            in_proj = self.in_projs[0]
+            assert in_proj.bias is None
+            assert isinstance(in_proj, nn.Linear)
+            dim = in_proj.weight.shape[0] // 3
```

Comment:
Cosmetic nit but you could have avoided the manual slicing via `qw, kw, vw = in_proj.weight.chunk(3)`.